posit-dev/positron
# Viewer pane pops out to editor tab

addresses #3714 

Example video (bonus content: if you turn volume up you can hear puppy snoring 💤 )
 
https://github.com/user-attachments/assets/9811ba80-9fd3-4bab-b12e-6d6376ebfe3c

Can pop out HTML or url from viewer pane to editor tab. Should work for multiple editors, independently of the Viewer pane content. 

Known problems/what is not included:

- cannot go directly from `Preview` button in Quarto file to editor. It seems like Quarto has a different toolbar than urls/html (same with html sent to output, eg, #4601)
- no ability to go from HTML file directly to editor
- no toggle as an experimental feature, but happy to add that in if we think its the right choice!

very inspired by #4364 🫶 

# QA notes

These are all gestures I've gone through and confirmed working! It looks like a lot of steps but includes a lot of opening up files 😄 

- Have 2 distinct`.qmd` files
- `quarto render filename_1.qmd` in Terminal so you have HTML output
- Open HTML in Viewer
- Click editor button (icon shown in video) to open in editor
- Resize newly opened editor, should not overlap
- With `filename_1.qmd` still in editor,`quarto preview filename_2.qmd` to get a `localhost` url
- Click on url and open in Viewer
- Click editor button so you have both `.qmd` files in editors
- Switch between editors, should reopen prior content
- Clear Viewer pane, should affect editors
- Close editors
